opentelemetry-sdk/tests/trace/export/test_export.py

```diff
@@ -340,6 +340,43 @@ def test_batch_span_processor_scheduled_delay(self):
 
         span_processor.shutdown()
 
+    def test_batch_span_processor_reset_timeout(self):
+        """Test that the scheduled timeout is reset on cycles without spans"""
+        spans_names_list = []
+
+        export_event = threading.Event()
+        my_exporter = MySpanExporter(
+            destination=spans_names_list,
+            export_event=export_event,
+            export_timeout_millis=50,
+        )
+
+        span_processor = export.BatchExportSpanProcessor(
+            my_exporter, schedule_delay_millis=50,
+        )
+
+        with mock.patch.object(span_processor.condition, "wait") as mock_wait:
+            _create_start_and_end_span("foo", span_processor)
+            self.assertTrue(export_event.wait(2))
+
+            # give some time for exporter to loop
+            # since wait is mocked it should return immediately
+            time.sleep(0.05)
+            mock_wait_calls = list(mock_wait.mock_calls)
+
+            # find the index of the call that processed the singular span
+            for idx, wait_call in enumerate(mock_wait_calls):
+                _, args, __ = wait_call
+                if args[0] <= 0:
```

Comment:
Wouldn't it be the only call to `wait()` that wasn't `wait(self.schedule_delay_millis / 1e3)`? Might be a little less flaky than relying on `time.sleep()` being accurate

Reply:
That's true, I can change it to check that only one call to `wait()` fits the criteria.

Admittedly I'm not the most familiar with testing multithreaded code in Python!